consensus/istanbul/core/core.go

```diff
@@ -37,13 +37,70 @@ import (
 	"github.com/syndtr/goleveldb/leveldb"
 )
 
+// IstBackendForCore provides the Istanbul backend application specific functions for Istanbul core
+type IstBackendForCore interface {
```

Comment:
what about "CoreBackend" as in "the necessary interface that Core needs to function" or "what core asks from others"